integration/lsp.test.ts

```diff
@@ -0,0 +1,125 @@
+function make_request(data) {
+    data.jsonrpc = "2.0";
+    data.id = 1;
+    const json = JSON.stringify(data);
+    return `Content-Length: ${json.length}\r\n\r\n${json}`;
+}
+
+function parse_response(data) {
+    let json = JSON.parse(data.split('\r\n')[2]);
+    delete json.id;
+    delete json.jsonrpc;
+    return json;
+}
+
+test('initialize', async () => {
+    import('@influxdata/flux-lsp-node')
+    .then(async ({Server}) => {
+        const expected = {
+            "result": {
+                "capabilities": {
+                    "textDocumentSync":1,
+                    "referencesProvider":true,
+                    "definitionProvider":true,
+                    "renameProvider":true,
+                    "foldingRangeProvider":true,
+                    "documentSymbolProvider":true,
+                    "documentFormattingProvider":true,
+                    "completionProvider": {
+                        "resolveProvider":true,
+                        "triggerCharacters": [".",":","(",",","\""]
+                    },
+                    "signatureHelpProvider": {
+                        "triggerCharacters": ["("],
+                        "retriggerCharacters": ["("]
+                    },
+                    "hoverProvider":true
+                }
+            }
+        };
+
+        const server = new Server(false, false);
+        const request = make_request({method: "initialize"});
+
+        const response = await server.process(request);
+        const error = response.get_error();
+        expect(error).toBe(undefined);
+        const message = parse_response(response.get_message());
+        expect(message).toStrictEqual(expected);
+    });
+});
+
+test('shutdown', () => {
+    import('@influxdata/flux-lsp-node')
+    .then(async ({Server}) => {
+        const expected = {"result": null };
+        const server = new Server(false, false);
```

Comment:
Since we're using the `Server` the same in most of these tests, it might be worth extracting to a higher scope so that it can be reused.

```node
describe('Server', async () => {
    const { Server } = await import('@influxdata/flux-lsp-node');
    const server = new Server();
    
    it('shuts down correctly', () => {
        server.process(req)
    });
    
    it('does another thing', () => {
        server.process(req)
    });
});
```

Reply:
The `Server` instances maintain some state, so they'll each have to be created, but yeah, that's a much better way to construct that. Making this change.

Reply:
This probably won't actually work because it looks like you can't use `async` with a `describe`, but you could still [do something similar to this using `beforeAll`](https://github.com/facebook/jest/issues/2235#issuecomment-426359880)

Reply:
Hm, I can't make this change, because `describe` can't take a promise. It wants all of its tests registered immediately.

Reply:
I'm not a huge fan of hoisting the state up to the test level, but in this case, it's probably not a huge deal. This has been addressed.